public/scripts/power-user.js

```diff
@@ -1070,8 +1070,8 @@ export function formatInstructModeChat(name, mes, isUser, isNarrator, forceAvata
     const separator = power_user.instruct.wrap ? '\n' : '';
     const separatorSequence = power_user.instruct.separator_sequence && !isUser
         ? power_user.instruct.separator_sequence
-        : (power_user.instruct.wrap ? '\n' : '');
-    const textArray = includeNames ? [sequence, `${name}: ${mes}`, separatorSequence] : [sequence, mes, separatorSequence];
+        : separator;
+    const textArray = includeNames ? [sequence, `${name}: ${mes}` + separatorSequence] : [sequence, mes + separatorSequence];
```

Comment:
**Extraneous newlines:** There's code that adds the separatorSequence, which is a newline by default when "Wrap Sequences with Newline" is on and the Separator isn't set explicitly. And then there's code that joins the lines with newlines by default when "Wrap Sequences with Newline" is on. Both together cause double newlines, so I've appended the separator directly without joining, fixing this issue. Please double-check this part, though, to make sure it doesn't have unintended side effects.